Introduction to C# and Setup for Visual Studio (#1)
LaunchCodeEducation/csharp-web-dev-curriculum: content/_index.md

date: 2021-10-01T09:28:27-05:00
draft: false
---

# C# Web Development

## What is this Course

This course is an introduction to object-oriented programming in C#. Students will then move on to building web applications with the industry-grade framework, ASP.NET Core MVC.

## Why take this Course

> This course is an introduction to **C#** and the .NET framework. C# is a programming language developed by Microsoft and used in almost all of its products.

Reviewer: Why is C# bolded here, but not other places?

Author: I fixed the repetition. C# is bolded because we always bold important terms the first time they are seen in the book, but not afterward.

Reviewer: Thanks Sally for the explanation.

> This course is designed for learners who are already familiar with at least one programming language. Thus, we’ll move quickly through the syntax rules and procedural basics of the language (ie, how does one write a `for` loop in C#? a conditional statement? Etc.).

Reviewer: ```suggestion
This course is designed for learners who are already familiar with at least one programming language. Thus, we’ll move quickly through the syntax rules and procedural basics of the language, i.e., how does one write a `for` loop in C# or a conditional statement?
```
```suggestion
This course is designed for learners who are already familiar with at least one programming language. Thus, we’ll move quickly through the syntax rules and procedural basics of the language (ie, how does one write a `for` loop in C#? a conditional statement? Etc.).
```
```suggestion
This course is designed for learners who are already familiar with at least one programming language. Thus, we’ll move quickly through the syntax rules and procedural basics of the language (ie, how does one write a `for` loop in C#? a conditional statement? Etc.).
```

As you’ve likely heard before, once you grasp the fundamentals of programming in one language, learning another becomes much easier. Professional software developers often work in environments involving several programming languages so it’s wise to start learning new ones early on in your career.

Beyond the basics of programming in C#, you’ll learn some key object oriented concepts to enhance the capabilities of your programs. You’ll also download a program called **Visual Studio IDE** to get familiar with the tools of the C# developer. In the final lessons of this course, you will use a C# framework called **ASP.NET Core MVC**. This framework gives us the scaffolding necessary to create MVC programs (another concept we’ll cover) effectively and efficiently.

## Where this Course Leads

Upon completing the content within this course you will be capable of building a web application with ASP.NET Core MVC that can store data in a local database. 

> After completing this course, you are ready to begin our Liftoff program.

Reviewer: Do students automatically enter LiftOff or is it opt in?  Might be nice to give a brief description about what they will learn in liftoff.

After completing this course is restating the sentence above -> You may continue your learning journey...

Author: Checked with Lin to make sure new language is appropriate but it sounds like it will be automatic in the future.

Reviewer: Looks good Sally, thanks for updating.
